[Docs] update android requirements instructions

Currently [the android instructions](https://github.com/koreader/koreader/blob/master/doc/Building_targets.md#for-android-1) say it requires `openjdk-8-jdk`, but actually wanting to compile `KOReader 2024.07 "Outside"` it complains it wants java 17, example (using arch linux `extra/jdk11-openjdk`):

```txt
FAILURE: Build failed with an exception.

* Where:
Build file '/mnt/projects/android/koreader/platform/android/luajit-launcher/app/build.gradle' line: 1

* What went wrong:
A problem occurred evaluating project ':app'.
> Failed to apply plugin 'com.android.internal.application'.
   > Android Gradle plugin requires Java 17 to run. You are currently using Java 11.
      Your current JDK is located in /usr/lib/jvm/java-11-openjdk
      You can try some of the following options:
       - changing the IDE settings.
       - changing the JAVA_HOME environment variable.
       - changing `org.gradle.java.home` in `gradle.properties`
```

--- a/doc/Building_targets.md
+++ b/doc/Building_targets.md
@@ -43,13 +43,13 @@ Some platforms will require additional packages:
 
 #### for Android
 
-Building for Android requires `openjdk-8-jdk` and `p7zip-full`.
+Building for Android requires `openjdk-17-jdk` and `p7zip-full`.
 
 
 For both Ubuntu and Debian, install the packages:
 
 ```
-sudo apt-get install openjdk-8-jdk p7zip-full
+sudo apt-get install openjdk-17-jdk p7zip-full
 ```
 
 #### for Debian

--- a/doc/Building_targets.md
+++ b/doc/Building_targets.md
@@ -43,13 +43,13 @@ Some platforms will require additional packages:
 
 #### for Android
 
-Building for Android requires `openjdk-17-jdk` and `p7zip-full`.
+Building for Android requires `openjdk-17-jdk-headless` and `p7zip-full`.
 
 
 For both Ubuntu and Debian, install the packages:
 
 ```
-sudo apt-get install openjdk-17-jdk p7zip-full
+sudo apt-get install openjdk-17-jdk-headless p7zip-full
 ```
 
 #### for Debian

